src/Serilog.Sinks.Logz.Io/LogzIoSerializer.cs

```diff
@@ -50,6 +53,27 @@ public LogzIoSerializer(LogzIoTextFormatterFieldNaming fieldNaming, bool include
         }
     }
 
+    /// <summary>
+    ///     Handle errors serializing the JSON event and log to <see cref="SelfLog"/> for diagnostics.
+    ///     <br /><br />
+    ///     This is particularly useful if certain properties throw exceptions, such sa properties on the logged
+    ///     <see cref="Exception"/>. Instead of losing the entire log event, only the problematic properties are lost.
+    /// </summary>
+    /// <param name="sender">The JSON serializer.</param>
+    /// <param name="e">Description of JSON serialization error.</param>
+    private static void OnJsonError(object sender, ErrorEventArgs e)
+    {
+        SelfLog.WriteLine("Failed to serialize log event for path {0}, member {1}, object {2}",
+            e.ErrorContext.Path, e.ErrorContext.Member, e.CurrentObject?.GetType().FullName);
+
+        if (e.ErrorContext.Error is { } exception)
+        {
+            SelfLog.WriteLine($"{exception.Message} {exception.StackTrace}");
+        }
+
+        e.ErrorContext.Handled = true;

```

Comment:
I can't think of a scenario where we don't want to mark as handled and leak the exception.